rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,124 @@
+Parameterized Metric Support
+============================
+
+Parameterized metrics are a subset of dynamic request metrics.  Specifically they focus on modifying the calculation of metrics by altering values used in the calculation of existing metrics or mapping their values at aggregation time.
+
+The expected contract of parameterized metrics is as follows: Given a base metric with api name `metricName`, it can be suffixed with a procedure-call list of keyword parameter names and values.  These values will be used in the calculation of the resulting new metric.
+
+So
+```
+"metricName(parameterName=parameterValue)"
+```
+
+will result in a dynamic metric based on `metricName`, but substituting the parameter value when calculating it.  Also, the above string will be used as the column name in the resulting report response schema. 
+
+Motivation
+----------
+
+Standard metrics in Fili are configured at load time.  However if reporting requires many different similar caluclations, such as adjusting precision or converting values of different units into a shared aggregate, or applying custom metric filters, it may be impractical or even impossible to configure all the interesting report metrics.
+
+With parameterized metrics, open-ended variations may be supported for a given conceptual metric calculation, simplifying configuration and possibly unlocking otherwise impractical freedom in metric calculation.
+
+
+The structure of a parameterized metric:
+----------------------------------------
+
+```
+metricName(parameterName=value)
+```
+
+metricName -> an api metric name which should appear as a resource at the `/v1/metrics/metricName` endpoint.
+
+Balanced outer parentheses are the signal that a metric is to be processed as parameterized metric.  Imbalanced parentheses or unescaped internal parentheses will probably result in request errors.
```

Comment:
This seems like a meaningless statement to me. There is (or should be) a grammar. Anything not conforming is an error. Balancing of parens is not the signal that the expression is good.